src/xcode/ENA/ENA/Source/Developer Menu/Features/DMSubmissionStateViewController.swift

```diff
@@ -0,0 +1,260 @@
+			if let error = error {
+				fatalError("err: \(error.localizedDescription)")
+			}
+			guard let localKeys = localKeys else {
+				fatalError("err")
+			}
+			print(localKeys)
+			self.client.fetchAllKeys { downloadedPackages in
+				print(downloadedPackages)
+				let allPackages = downloadedPackages.allKeyPackages
+				let allRemoteKeys = Array(allPackages
+					.compactMap { try? $0.keys() }
+					.joined()
+				)
+
+				var foundKeys = [ENTemporaryExposureKey]()
+				var missingKeys = [ENTemporaryExposureKey]()
+
+
+				for localKey in localKeys {
+					let found = allRemoteKeys.containsKey(localKey)
+					if found {
+						foundKeys.append(localKey)
+					} else {
+						missingKeys.append(localKey)
+					}
+				}
+				self.checkResult = .init(
+					missingKeys: missingKeys,
+					foundKeys: foundKeys
+				)
+				self.tableView.reloadData()
+			}
+		})
+	}
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		2
+	}
+
+	override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+		switch section {
+		case 0:
+			return "Missing Keys \(checkResult.missingKeys.count)"
+		case 1:
+			return "Found Keys \(checkResult.foundKeys.count)"
+		default: fatalError("fail")
+		}
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		switch section {
+		case 0:
+			return checkResult.missingKeys.count
+		case 1:
+			return checkResult.foundKeys.count
+		default: fatalError("fail")
+		}
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableKeyCell(for: indexPath)
+		let key: ENTemporaryExposureKey
+		let row = indexPath.row
+		switch indexPath.section {
+		case 0:
+			key = checkResult.missingKeys[row]
+		case 1:
+			key = checkResult.foundKeys[row]
+		default: fatalError("fail")
+		}
+
+		cell.configure(
+			with: DMKeyCell.Model(
+				keyData: key.keyData,
+				rollingStartIntervalNumber: Int32(key.rollingStartNumber),
+				transmissionRiskLevel: Int32(key.transmissionRiskLevel)
+			)
+		)
+		return cell
+	}
+}
+
+private extension Data {
+	static let binHeader = Data("EK Export v1    ".utf8)
+	var withoutBinHeader: Data {
+		let headerRange = startIndex ..< Data.binHeader.count
+
+		guard subdata(in: headerRange) == Data.binHeader else {
+			return self
+		}
+		return subdata(in: headerRange.endIndex ..< endIndex)
+	}
+}
+
+extension SAPDownloadedPackage {
+	var binProtobufData: Data {
+		bin.withoutBinHeader
+	}
+
+	func keys() throws -> [Apple_TemporaryExposureKey] {
+		let data = binProtobufData
+		let export = try Apple_TemporaryExposureKeyExport(serializedData: data)
+		return export.keys
+	}
+}
+
+private extension Array where Element == Apple_TemporaryExposureKey {
+	func containsKey(_ key: ENTemporaryExposureKey) -> Bool {
+		contains { appleKey in
+			appleKey.keyData == key.keyData
+		}
+	}
+}
+
+extension Client {
```

Comment:
I made the extension private now since it is only used in this view controller. It should not be available to the app-related code. Do you agree with simply making it private?